server/server.go

```diff
@@ -535,13 +556,16 @@ func (server HTTPServer) reportEndpoint(writer http.ResponseWriter, request *htt
 		rules = append(rules, rule)
 	}
 
-	if len(aggregatorResponse.Report) != 0 && len(rules) == 0 {
+	if len(aggregatorResponse.Report) != 0 && !hasNoErrors {
```

Comment:
why?

Reply:
Because `len(rules)` could be empty because of filtering, but also because of `errors`, so this `if` should be triggered only in case of errors.

Reply:
but you could write an error earlier, when it happens in the loop and stop the whole thing, you ignore rules anyways

Reply:
btw, do you really need to return an error if this condition `if OSDEligible && !ruleWithContent.NotRequireAdmin {` is met only for one rule? Maybe just ignore this rule

Reply:
Yes, and no, because if we handle errors in this place, will mean, that in logs we get all rules, that's couldn't be parsed or fetched, which is much more better than stop request after first error.

Reply:
It's not returning error when it's OSD, it just return not successful state, which mean that this rule will be ignored.

Reply:
Now you changed the logic and the error will be returned to the user if a single rule couldn't parse, not sure if we want this logic, but previous version was different(we just logged this error and returned what we could). @tisnik wdyt?